switch to graph freezing and allow users to specify output node names

tjingrant opened the pull request.

fumihwh commented on onnx_tf/converter.py:
```diff
@@ -166,20 +174,47 @@ def convert(infile, outfile, convert_to, **kwargs):
+          initializer_nodes="")
+
+      # Load back the frozen graph.
+      with open(workdir + "/frozen_model.pb", "rb") as f:
```
Should we remove intermediate products?

tjingrant replied:
Yeah we probably should.

tjingrant commented:
@fumihwh summary of changes:
1. Allow onnx-tf to ignore purely training related ops like queue shuffling.
2. Automatically translate QueueDequeueManyV2 to an input.
3. Allow users to specify inference graph (useful for dropout removal, we should eventually automate dropout removal for inference).
4. Refactored slightly. Also addressed ur comment.

fumihwh commented on onnx_tf/common/data_type.py:
```diff
@@ -55,3 +55,18 @@ def _onnx_dtype(dtype):
+
+
+# TODO (tjingrant) unify _onnx_dtype into any_dtype_to_onnx_dtype
+def any_dtype_to_onnx_dtype(np_dtype, tf_dtype, onnx_dtype):
```
Is it OK that all three are positional args?

tjingrant replied:
I'm not sure what is your concern, whenever I use any_dtype_to_onnx_dtype, I always use them as keyword arguments.

fumihwh commented on onnx_tf/frontend.py:
```diff
@@ -84,6 +100,7 @@ def tensorflow_graph_to_onnx_graph(cls,
               node_dict=dict(node_tup),
               data_type_cast_map=onnx_graph.data_type_cast_map)
         else:
+          print(node.op_type)
```
Dev code?
